### PR TITLE
Cherry-pick a variety of changes into 1.4.1 (Continuous Integration)

### PR DESCRIPTION
**Pull request by zjs**
(Note: This _must_ be submitted with the "rebase & merge" option, _not_ "squash & merge".)

Cherry-pick a variety of work from `master` to `releases/1.4.1` to include in 1.4.1-rc1. This should be a series of independent PRs, but I'm posting this all at once to allow a quicker turn-around cycle.

Cherry picks:
 * ac9d15b18 from #6893
 * 0ff435cd2 from #7727
 * 4f32db225 from #8003 
 * 7dd6fc328 from #7896
 * 584e96a7 from #8082

`[full ci]`

### CODE REVIEW

File: tests/integration-test.sh [hunk 1]
```diff
@@ -84,7 +84,14 @@ echo "default_project_id = $GS_PROJECT_ID" >> $botofile
 
 
 if [ -f "$outfile" ]; then
-  gsutil cp $outfile gs://vic-ci-logs
+  echo `ls -al $outfile`
```
Comment (hickeng) [inline tests/integration-test.sh]: I'm not clear how this is different to (`echo` seems irrelevant):
```
ls -al $outfile
```

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot [hunk 2]
```diff
@@ -54,10 +61,36 @@ Inspect VCH Config ${name}
...
     ${id}=    Run    echo '${OUTPUT}' | jq -r '.vchs[] | select(.name=="${name}").id'
+    Set Test Variable    ${id}
 
     Get Path Under Target    vch/${id}
 
 
+Pull Busy Box
+    [Arguments]    ${docker_params}    ${expected_rc}    ${expected_output}
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker -H ${docker_params} pull busybox
```
Comment (hickeng) [inline tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: should be `${busybox}`

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot [hunk 3]
```diff
@@ -54,10 +61,36 @@ Inspect VCH Config ${name}
...
+
+    Should Be Equal As Integers    ${rc}    ${expected_rc}
+    Should Contain    ${output}    ${expected_output}
+
+
+Get Docker Host Params
+    [Arguments]    ${vch_name}
+    Run Keyword And Continue On Failure    Wait Until Keyword Succeeds    30x    10s    Get Docker Params API    ${vch_name}
+    Run Keyword If    "${docker_host}" == "null"    Run VIC Machine Inspect Command   ${vch_name}
```
Comment (hickeng) [inline tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: do we expect the `Get Docker Params API` to fail? If it's fault injection, ok... but I don't see it here.
Get it from both sources and check they're the same... that's a viable test.

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot [hunk 4]
```diff
@@ -90,7 +123,7 @@ Create minimal VCH
     Property Should Contain         .runtime.power_state                 poweredOn
     Property Should Contain         .runtime.upgrade_status              Up to date
 
-    Wait For VCH Initialization  name=%{VCH-NAME}-api-test-minimal
+    Get Docker Host Params    %{VCH-NAME}-api-test-minimal
```
Comment (hickeng) [inline tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: Still looks like it should be waiting for VCH initialization but looks like that's been replaced with `Wait Until Keyword Succeeds` for 300s.

File: tests/integration-test.sh (hunk 1 above)
Comment (zjs) [reply tests/integration-test.sh]: #8086

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot (hunk 2 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: I believe we actually want `busybox` in this case, since `${busybox}` comes from harbor (https://github.com/vmware/vic/pull/7727#discussion_r181495422).

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot (hunk 3 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: This was an intention decision by the author of the test, as discussed here: https://github.com/vmware/vic/pull/7727#discussion_r181500754

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot (hunk 4 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: Related to https://github.com/vmware/vic/pull/8079/files/54c60585b6dc1bd225502b7a82b7cd453279bbaa#r197295633

**Comment by sflxn** [comment]
I added two comments about adding comments for things that were in the merged PRs' comment sections, but I understand this wasn't your merged PRs.

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot (hunk 2 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: Added on master: #8089

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot (hunk 3 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: Added on master: #8089
